controller/api/destination/watcher/cluster_store_test.go

```diff
@@ -0,0 +1,262 @@
+package watcher
+
+import (
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/controller/k8s"
+)
+
+func CreateMockDecoder() configDecoder {
+	// Create a mock decoder with some random objs to satisfy client creation
+	return func(data []byte, enableEndpointSlices bool) (*k8s.API, *k8s.MetadataAPI, error) {
+		remoteAPI, err := k8s.NewFakeAPI([]string{}...)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		metadataAPI, err := k8s.NewFakeMetadataAPI(nil)
```

Comment:
Following up on our off-github convo, I think the race occurs between these two calls `NewFakeAPI` and `NewFakeMetadataAPI`, because they rely on a call to `ToRuntimeObject()` which uses this global struct instance from `scheme/register.go`, which holds the map that is contended over:
```go
var Scheme = runtime.NewScheme()
```
Maybe a simple fix could be to have `ToRuntimeObject()` receive as an argument the scheme instance, which could be instantiated separately for these two API clients.